content/en/blog/2022-03-25-putting-the-git-into-gitops/index.md

```diff
@@ -0,0 +1,212 @@
+`openssl`/`libssh2` packages of different Linux distributions act in
+[slightly different
+ways](https://github.com/fluxcd/golang-with-libgit2/blob/libgit2-1.3.0/hack/Makefile#L63-L69).
+This created a yet different problem: the versions we shipped on the
+containers could behave in different ways when we were developing on our
+Mac/Linux machines. This forced us to cross-compile statically built
+libraries that we can simply download at both development time or
+statically linking them into the final binary we create when releasing
+our controllers.
+
+We decided to build the libraries for the AMD64, ARM64 and ARMv7
+architectures and link them statically, which was a prerequisite for us
+to [enable
+fuzzing](/blog/2022/02/security-more-confidence-through-fuzzing/)
+for all Flux controllers. Getting this all up and running for every
+upstream release and making sure it's all covered nicely with tests is a
+challenge and an area of work we invested quite a lot of time in.
+
+## What's next in Git things?
+
+`libgit2` does not expose concepts to allow users to set timeouts for
+network operations, meaning that most git operations could hang
+indefinitely in specific circumstances. This would result in specific
+`GitRepository` objects getting stuck and stop updating until the
+controllers get restarted - Users reported this for both the
+image-automation and source controllers in the past 6 months.
+
+We had a few challenges getting traction making changes upstream to fix
+similar issues, so to avoid delaying a fix or forking the dependency, we
+decided to add experimental support for Go managed transports, which means
+we can enforce that network operations won't take more than a given
+amount of time to complete, but without requiring any changes upstream.
+
+This is [part of Flux 0.28](https://github.com/fluxcd/flux2/releases/tag/v0.28.0)
+and can be enabled by adding an environment `EXPERIMENTAL_GIT_TRANSPORT=true`
+in both `source` and `image-automation` controllers.
+
+This will give us more control over the transport with Go native
+transport using the `libgit2` smart transport support. Read the
+[source-controller
+changelog](https://github.com/fluxcd/source-controller/blob/main/CHANGELOG.md#experimental-managed-transport-for-libgit2-git-implementation)
+for more information.
+
+If you want to enable this automatically, just add the following to
+your `kustomization.yaml`:
+
+```yaml
+patches:
+- patch: |
+    - op: add
+      path: /spec/template/spec/containers/0/env/0
+      value:
+        name: EXPERIMENTAL_GIT_TRANSPORT
+        value: "true"
+  target:
+    kind: Deployment
+    name: "(source-controller|image-automation-controller)"
+```
+
+We are also adding support for sha256 hashes: Git supports it since 2.29
+but all major Git service provides such as GitLab and GitHub still don't.
+We are watching this to be able to adopt support as the industry moves
+on from sha1.
```

Comment:
```suggestion
Support for sha256 hashes: Git CLI supports it since 2.29, however all major Git
service providers, such as GitLab and GitHub, are yet to make some progress on 
this front.
Upstream, libgit2 is starting to pave the way for that support on [v1.4.0]https://github.com/libgit2/libgit2/releases/tag/v1.4.0),
we are watching this space to be able to support Flux users as the industry moves on
from SHA1.
```